src/app/public/modules/dock/dock.service.ts

```diff
@@ -72,8 +92,35 @@ export class SkyDockService {
     return item;
   }
 
```

Comment:
Since `SkyDockService` is a singleton, it seems like these options would spillover between components.

Reply:
Yep. If you are using the singleton service they could. I'll write some docs about this in the JS Docs.

The trick here is that we changed this to be provided in `root` intentionally. This allows for a singleton for most dock things. And then, where appropriate, consumers can overwrite this singleton by using a component provider in order to get another instance of the service outside the main singleton. You can see this in practice in my modals PR that I created yesterday. The idea is that we then have an option to have separate docks where needed while the service itself is still tied to a single dock (as was already the case with the `insertComponent` function above). 

This was the approach we had settled on in triad prior to starting this work. I'll get some docs added here though as I had missed them and I can call out the singleton nature in them.

Reply:
Added docs